### ISSUE TITLE
attention_map getting issue

### ISSUE THREAD

**Issue by 838959823** [jinyeying/DC-ShadowNet-Hard-and-Soft-Shadow-Removal#23]
I encountered two problems in reproducing the author's work：
 1. Using the provided pre-trained model and test code, the obtained attention map does not focus on the shadow region
 2. The attention map mentioned in the paper is the mean value of the weighted multi-channel feature map, while in the code it is torch.sum(x,dim=1)
I don't know if I'm not doing it correctly, hope someone can help me

**Comment by jinyeying** [comment]
> I encountered two problems in reproducing the author's work：
> 
> 1. Using the provided pre-trained model and test code, the obtained attention map does not focus on the shadow region
> 2. The attention map mentioned in the paper is the mean value of the weighted multi-channel feature map, while in the code it is torch.sum(x,dim=1)
>    I don't know if I'm not doing it correctly, hope someone can help me

1. Our attention visualization is in the training process.
2. I think you are correct, the equation is $\sum_{i=1}^{n}{\mathbf{w}^g_s}_i {\mathbf{\Pi}^g_s}_i$, which sums up all the weighted feature maps.

**Comment by jinyeying** [comment]
> I encountered two problems in reproducing the author's work：
> 
> 1. Using the provided pre-trained model and test code, the obtained attention map does not focus on the shadow region
> 2. The attention map mentioned in the paper is the mean value of the weighted multi-channel feature map, while in the code it is torch.sum(x,dim=1)
>    I don't know if I'm not doing it correctly, hope someone can help me

But I think it does not matter, you can try torch.mean, but seems differences are small.
